scripts/github-actions/generate-module-index-data.js

```diff
@@ -98,13 +117,114 @@ async function generateModuleIndexData({ require, github, context, core }) {
     }
   }
 
+  // AVM Resource Modules
+  for (const moduleGroup of moduleGroupsAvmRes) {
+    const moduleGroupPath = path.join("avm/res", moduleGroup);
+    const moduleNames = await getSubdirNames(fs, moduleGroupPath);
+
+    for (const moduleName of moduleNames) {
+      const modulePath = `avm/res/${moduleGroup}/${moduleName}`;
+      const versionListUrl = `https://mcr.microsoft.com/v2/bicep/${modulePath}/tags/list`;
+      const moduleBicepRegistryRefSplit = modulePath
+        .split(/[\/\\]avm[\/\\]/)
+        .pop();
+      const moduleBicepRegistryRefReplace = moduleBicepRegistryRefSplit
+        .replace(/-/g, "")
+        .replace(/[\/\\]/g, "-");
+      const moduleBicepRegistryRef = moduleBicepRegistryRefReplace;
+
+      try {
+        core.info(`Processing AVM Resource Module ${modulePath}:...`);
+        core.info(`  Retrieving AVM Resource Module ${versionListUrl}`);
+        core.info(`    Git Tag: ${modulePath}`);
+        core.info(`    BRM Ref: ${moduleBicepRegistryRef}`);
+
+        const versionListResponse = await axios.get(versionListUrl);
+        const tags = versionListResponse.data.tags.sort();
+
+        const properties = {};
+        for (const tag of tags) {
+          var description = await getModuleDescription(
+            github,
+            core,
+            path,
+            modulePath,
+            (moduleRoot = "avm/res"),
+            tag,
+            context
+          );
+          if (description) {
+            properties[tag] = { description };
+            modulesWithDescriptions[modulePath] = true;
+          }
+        }
+
+        moduleIndexData.push({
+          moduleName: moduleBicepRegistryRef,
+          tags,
+          properties,
+        });
+      } catch (error) {
+        core.setFailed(error);
+      }
+    }
+  }
+
+  // AVM Pattern Modules
+  for (const moduleName of moduleGroupsAvmPtn) {
```

Comment:
[non-blocking]: The two loops for processing AVM modules are almost identical (they are similar to brm loop as well). It would be good if we could extract the shared logic in a helper function to remove duplicates. This can be done in the future PR though.

Reply:
completely agree, will take a go at this in a new PR